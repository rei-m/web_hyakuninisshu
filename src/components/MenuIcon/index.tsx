import * as React from 'react';
import styled from 'styled-components';

const Root = styled.div`
  padding: 16px;
`;

const Icon = styled.i`
  width: 150px;
  height: 150px;
  margin-bottom: 30px;
  border: 4px solid;
  border-radius: 50%;
  color: #f1b400;
  font-size: 6rem;
  line-height: 142px;
  text-align: center;
`;

export enum IconType {
  Training,
  Exam,
  Material
}

export interface MenuIconProps {
  readonly iconType: IconType;
}

interface Value {
  readonly icon: string;
  readonly title: string;
  readonly explain: string;
}

const TRAINING_VALUE: Value = {
  icon: 'create',
  title: '練習',
  explain: '説明',
};

const EXAM_VALUE: Value = {
  icon: 'note',
  title: '力試し',
  explain: '説明',
};

const MATERIAL_VALUE: Value = {
  icon: 'library_books',
  title: '資料',
  explain: '説明',
};

const iconTypeToValue = (iconType: IconType) => {
  switch(iconType) {
    case IconType.Training:
      return TRAINING_VALUE;
    case IconType.Exam:
      return EXAM_VALUE;
    case IconType.Material:
      return MATERIAL_VALUE;
    default:
      throw new Error('unknown TypeName');
  }
};

const MenuIcon = (props: MenuIconProps) => {
  const value = iconTypeToValue(props.iconType);
  return (
    <Root>
      <Icon className="material-icons">{value.icon}</Icon>
      <div>{value.title}</div>
      <p>{value.explain}</p>
    </Root>
  );
}

export default MenuIcon;
