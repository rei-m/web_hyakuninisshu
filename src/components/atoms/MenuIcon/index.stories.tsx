import React from 'react';
import { storiesOf } from '@storybook/react';
import { ExamIcon, MaterialIcon, OtherIcon, TrainingIcon } from './index';

storiesOf('atoms/MenuIcon', module)
  .add('TrainingIcon', () => <TrainingIcon />)
  .add('ExamIcon', () => <ExamIcon />)
  .add('MaterialIcon', () => <MaterialIcon />)
  .add('OtherIcon', () => <OtherIcon />);
