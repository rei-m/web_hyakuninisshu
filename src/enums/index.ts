// このIFは苦肉の策だが他にいい方法はあるのか。。。
export interface EnumCollection<T> {
  readonly values: T[];
  readonly valueOf: (value: T) => T;
}

export enum MenuType {
  Training,
  Exam,
  Material,
  Other
}

export enum QuestionState {
  InAnswer,
  Answered,
  ConfirmCorrect,
  Finished
}

export enum RangeFromCondition {
  One = 1,
  Eleven = 11,
  TwentyOne = 21,
  ThirtyOne = 31,
  FortyOne = 41,
  FiftyOne = 51,
  SixtyOne = 61,
  SeventyOne = 71,
  EightyOne = 81,
  NinetyOne = 91
}

export const RangeFromConditions: EnumCollection<RangeFromCondition> = {
  valueOf: (value: number) => {
    const result = RangeFromConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeFrom, value is ${value}`);
  },
  values: [
    RangeFromCondition.One,
    RangeFromCondition.Eleven,
    RangeFromCondition.TwentyOne,
    RangeFromCondition.ThirtyOne,
    RangeFromCondition.FortyOne,
    RangeFromCondition.FiftyOne,
    RangeFromCondition.SixtyOne,
    RangeFromCondition.SeventyOne,
    RangeFromCondition.EightyOne,
    RangeFromCondition.NinetyOne
  ]
};

export enum RangeToCondition {
  Ten = 10,
  Twenty = 20,
  Thirty = 30,
  Forty = 40,
  Fifty = 50,
  Sixty = 60,
  Seventy = 70,
  Eighty = 80,
  Ninety = 90,
  OneHundred = 100
}

export const RangeToConditions: EnumCollection<RangeToCondition> = {
  valueOf: (value: number) => {
    const result = RangeToConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeTo, value is ${value}`);
  },
  values: [
    RangeToCondition.Ten,
    RangeToCondition.Twenty,
    RangeToCondition.Thirty,
    RangeToCondition.Forty,
    RangeToCondition.Fifty,
    RangeToCondition.Sixty,
    RangeToCondition.Seventy,
    RangeToCondition.Eighty,
    RangeToCondition.Ninety,
    RangeToCondition.OneHundred
  ]
};

export enum KimarijiCondition {
  None = 0,
  One = 1,
  Two = 2,
  Three = 3,
  Four = 4,
  Five = 5,
  Six = 6
}

export const KimarijiConditions: EnumCollection<KimarijiCondition> = {
  valueOf: (value: number) => {
    const result = KimarijiConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Kimariji, value is ${value}`);
  },
  values: [
    KimarijiCondition.None,
    KimarijiCondition.One,
    KimarijiCondition.Two,
    KimarijiCondition.Three,
    KimarijiCondition.Four,
    KimarijiCondition.Five,
    KimarijiCondition.Six
  ]
};

export enum ColorCondition {
  None = '',
  Blue = 'blue',
  Pink = 'pink',
  Yellow = 'yellow',
  Green = 'green',
  Orange = 'orange'
}

export const ColorConditions: EnumCollection<ColorCondition> = {
  valueOf: (value: string) => {
    const result = ColorConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Color, value is ${value}`);
  },
  values: [
    ColorCondition.None,
    ColorCondition.Blue,
    ColorCondition.Pink,
    ColorCondition.Yellow,
    ColorCondition.Green,
    ColorCondition.Orange
  ]
};

export enum KarutaStyleCondition {
  KanjiAndKana = 0,
  KanaOnly = 1
}

export const KarutaStyleConditions: EnumCollection<KarutaStyleCondition> = {
  valueOf: (value: number) => {
    const result = KarutaStyleConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown KarutaStyle, value is ${value}`);
  },
  values: [KarutaStyleCondition.KanjiAndKana, KarutaStyleCondition.KanaOnly]
};
