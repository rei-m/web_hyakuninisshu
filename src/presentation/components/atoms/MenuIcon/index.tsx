import CreateIcon from '@material-ui/icons/Create';
import NoteIcon from '@material-ui/icons/Note';
import LibraryBooksIcon from '@material-ui/icons/LibraryBooks';
import SettingsIcon from '@material-ui/icons/Settings';

export const TrainingIcon = CreateIcon;
export const ExamIcon = NoteIcon;
export const MaterialIcon = LibraryBooksIcon;
export const OtherIcon = SettingsIcon;
