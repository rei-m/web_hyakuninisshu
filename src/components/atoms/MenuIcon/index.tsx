import CreateIcon from '@material-ui/icons/Create';
import NoteIcon from '@material-ui/icons/Note';
import LibraryBooksIcon from '@material-ui/icons/LibraryBooks';
import SettingsIcon from '@material-ui/icons/Settings';
import { MenuType } from '@src/enums';

export const TrainingIcon = CreateIcon;
export const ExamIcon = NoteIcon;
export const MaterialIcon = LibraryBooksIcon;
export const OtherIcon = SettingsIcon;

export const menuIcon = (iconType: MenuType) => {
  switch (iconType) {
    case MenuType.Training:
      return TrainingIcon;
    case MenuType.Exam:
      return ExamIcon;
    case MenuType.Material:
      return MaterialIcon;
    case MenuType.Other:
      return OtherIcon;
  }
};
