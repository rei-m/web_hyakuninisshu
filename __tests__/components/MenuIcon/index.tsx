import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import MenuIcon from '@src/components/MenuIcon';
import { MenuType } from '@src/enums';

describe('<MenuIcon />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(
      <MenuIcon iconType={MenuType.Exam} />
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
