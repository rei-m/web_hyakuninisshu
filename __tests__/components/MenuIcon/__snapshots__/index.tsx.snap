// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon /> should render component 1`] = `
<i
  className="material-icons sc-bdVaJa gXQMCH"
>
  note
</i>
`;
